frontend/src/pages/JobsListingPage.js

```diff
@@ -1,36 +1,41 @@
 import React, { useState, useEffect } from 'react';
-import * as jobsApi from '../mock-api/jobsApi';
+import { connect } from 'react-redux';
+import { getJobs } from '../actions';
 
+import { useHttpClient } from '../shared/hooks/http-hook';
 import SearchBox from '../components/SearchBox';
-
 import FilteringSection from '../components/FilteringSection';
 
-export default function JobsListingPage(props) {
-  const [jobs, setJobs] = useState([]);
-  const [searchField, setSearchField] = useState('');
+function JobsListingPage(props) {
+    const jobs = props.jobs;
+    const [searchField, setSearchField] = useState('');
+    const { isLoading, error, sendRequest, clearError } = useHttpClient();
 
-  useEffect(() => {
-    if (jobs.length === 0)
-      jobsApi.getJobs().then((jobsList) => {
-        setJobs(jobsList);
-      });
-  }, [jobs.length]);
+    useEffect(() => {
+        props.getJobs(sendRequest);
+    }, []);
 
-  const onSearchChange = (event) => {
-    setSearchField(event.target.value);
-  };
+    const onSearchChange = (event) => {
+        setSearchField(event.target.value);
+    };
 
-  if (jobs.length === 0) {
-    return <h1>Loading...</h1>;
-  } else {
-    return (
-      <div className="mx-auto">
-        <h1 className="text-center m-3">Jobs Page</h1>
-        <SearchBox className="" searchChange={onSearchChange} />
-        <div className="d-block">
-          <FilteringSection props={{ jobs, searchField }} />
-        </div>
-      </div>
-    );
-  }
```

Comment:
You could use ternary operator instead of {if ... else}. -> !jobs.length ? Loading : Rest